File: Features/dockerform-yocordapp/workflows/src/test/java/net/corda/samples/dockerform/flows/FlowTests.java
package net.corda.samples.dockerform.flows;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.transactions.SignedTransaction;
import net.corda.testing.node.MockNetwork;
import net.corda.testing.node.MockNetworkParameters;
import net.corda.testing.node.StartedMockNode;
import net.corda.testing.node.TestCordapp;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters().withCordappsForAllNodes(ImmutableList.of(
            TestCordapp.findCordapp("net.corda.samples.dockerform.contracts"),
            TestCordapp.findCordapp("net.corda.samples.dockerform.flows")
    )));
    private final StartedMockNode a = this.network.createNode();
    private final StartedMockNode b = this.network.createNode();

    public FlowTests() {
        ImmutableList.of(this.a, this.b).forEach(it -> {
            it.registerInitiatedFlow(YoFlowResponder.class);
        });
    }

    @Before
    public void setup() {
        this.network.runNetwork();
    }

    @After
    public void tearDown() {
        this.network.stopNodes();
    }

    //The yo flow should not have any input
    //This test will check if the input list is empty
    @Test
    public void dummyTest() throws ExecutionException, InterruptedException {
        CordaFuture<SignedTransaction> future = this.a.startFlow(new YoFlow(this.b.getInfo().getLegalIdentities().get(0)));
        this.network.runNetwork();
        SignedTransaction ptx = future.get();
        assert (ptx.getTx().getInputs().isEmpty());
    }
}

File: kotlin-samples/Basic/tutorial-applestamp/workflows/src/test/kotlin/com/tutorial/FlowTests.kt
package com.tutorial

import com.tutorial.flows.Initiator
import com.tutorial.states.TemplateState
import net.corda.core.node.services.Vault.StateStatus
import net.corda.core.node.services.vault.QueryCriteria
import net.corda.core.transactions.SignedTransaction
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkParameters
import net.corda.testing.node.StartedMockNode
import net.corda.testing.node.TestCordapp
import org.junit.After
import org.junit.Before
import org.junit.Test
import java.util.concurrent.Future


class FlowTests {
    private lateinit var network: MockNetwork
    private lateinit var a: StartedMockNode
    private lateinit var b: StartedMockNode

    @Before
    fun setup() {
        network = MockNetwork(
            MockNetworkParameters(
                cordappsForAllNodes = listOf(
                    TestCordapp.findCordapp("com.tutorial.contracts"),
                    TestCordapp.findCordapp("com.tutorial.flows")
                )
            )
        )
        a = network.createPartyNode()
        b = network.createPartyNode()
        network.runNetwork()
    }

    @After
    fun tearDown() {
        network.stopNodes()
    }

    @Test
    fun `DummyTest`() {
        val flow = Initiator(b.info.legalIdentities[0])
        val future: Future<SignedTransaction> = a.startFlow(flow)
        network.runNetwork()

        //successful query means the state is stored at node b's vault. Flow went through.
        val inputCriteria: QueryCriteria = QueryCriteria.VaultQueryCriteria().withStatus(StateStatus.UNCONSUMED)
        val state = b.services.vaultService.queryBy(TemplateState::class.java, inputCriteria).states[0].state.data
    }
}